tests/validator_test.go

```diff
@@ -320,13 +324,415 @@ var _ = Describe("Template validator", func() {
+				"vm.kubevirt.io/template":           templateWithoutRules.Name,
+				"vm.kubevirt.io/template-namespace": templateWithoutRules.Namespace,
+				"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:5035]: Template with validations, VM with validations", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("64M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 5,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           templateWithRules.Name,
+				"vm.kubevirt.io/template-namespace": templateWithRules.Namespace,
+				"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:5036]: should successfully create a VM based on the VM validation rules", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("64M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 5,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           templateWithRules.Name,
+				"vm.kubevirt.io/template-namespace": templateWithRules.Namespace,
+				"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 6
+        										 }
+												]`,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:5174]: VM with validations and deleted template", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("64M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 3,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           "fedora-desktop-small-with-rules",
```

Comment:
If the template does not exist, use a name that obviously does not exist, like `nonexisting-vm-template`